### PR TITLE
ci: add ci.yaml for testing on PRs

### PR DESCRIPTION
**Pull request by second-ed**
yaml to hopefully make it build and run the tests on PRs to main and dev branches

### CODE REVIEW

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+      - name: Setup Gradle
+        uses: gradle/actions/setup-gradle@v3
+      - name: Build
+        run: |
+          ./gradlew clean build
+      - name: Unit testing
```
Comment (PPartisan) [inline .github/workflows/ci.yaml]: I believe `./gradlew build` will also run the unit tests, so we can remove this step

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+name: CI
+
+on:
+  pull_request:
+    branches:
+      - "main"
+      - "dev*"
```
Comment (PPartisan) [inline .github/workflows/ci.yaml:7]: We don't actually have a development branch

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (second-ed) [reply .github/workflows/ci.yaml]: sweet, done as of [16e16e9](https://github.com/PPartisan/RockPaperScissorsTests/pull/3/commits/16e16e9f35443a8fc908d4ec27de86a217ef4376)

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (second-ed) [reply .github/workflows/ci.yaml:7]: I think it’s worth having that pattern as it future-proofs for when we move into a dev-test-main pattern or similar

**Review by PPartisan — APPROVED** [review]

**Pull request merged**